### ISSUE TITLE
Generated icon no longer used when building with com.android.tools.build:gradle:7.3.0-rc01

### ISSUE THREAD

**Issue by litrik** [usefulness/easylauncher-gradle-plugin#349]
After upgrading com.android.tools.build:gradle from 7.3.0-beta05 to 7.3.0-rc01, the easylauncher icons are nog longer used. They are still being generated in the build/generated/easylauncher folder and Android Studio still renders them as the actual launcher icon, but after building and deploying the app, the undecorated base icon is shown.

I'd be surprised that this is a bug in Easylauncher, probably Google broke something in version 7.3.0-rc01 of the Android Build tools.

**Comment by mateuszkwiecinski** [comment]
Oh, thanks for letting me know! I only caught the plugin is completely broken on `7.4.0-alpha10` and reported it here: https://issuetracker.google.com/issues/244063664
It seems like AGP changed its behavior in 7.3.0 too, but it fails silently there. I'll investigate and post an update on the issuetracker :)

**Comment by itolianezzz** [comment]
Hello guys! Any update here?
Android Studio Dolphin released with AGP 7.3.0 and generated icons are not used when building the app 😢

**Comment by miduch** [comment]
same here AGP 7.3.0

**Comment by emartynov** [comment]
I just wonder - are adaptive icons affected or only mipmap versions?

**Comment by jumpbayern** [comment]
I have the same error with AGP 7.3.0

**Comment by Jacks0N23** [comment]
Same
anyone found workaround?

**Comment by RustamSitdikov** [comment]
Please help with this issue

**Comment by huextrat** [comment]
Hi, @mateuszkwiecinski any news on this issue ? Or any functional workaround/alternatives ?

**Comment by litrik** [comment]
I can confirm that upgrading to `com.android.tools.build:gradle:7.4.0-beta02` and setting `android.disableResourceValidation=true` in `gradle.properties` does solve/workaround the issue.

**Comment by NitroG42** [comment]
But unfortunatelly it doesn't seems to work with 7.3.0

**Comment by ba-a** [comment]
I can't confirm the workaround. The badge is still not showing up for me. I tried as well with 
`com.android.tools.build:gradle:7.4.0-beta02` 
`android.disableResourceValidation=true` 
and AndroidStudio Electric Eel to make the gradle version 7.4.0 work. 
@litrik  is there anything else you changed?

**Comment by litrik** [comment]
> I can't confirm the workaround. The badge is still not showing up for me. I tried as well with `com.android.tools.build:gradle:7.4.0-beta02` `android.disableResourceValidation=true` and AndroidStudio Electric Eel to make the gradle version 7.4.0 work. @litrik is there anything else you changed?

Those were indeed the only changes I did to get it working with 7.4.0-beta02 (in Electric Eel and after a good Gradle sync, clean and rebuild of the project).

**Comment by casperbang** [comment]
I have the same issue as reported above. Sticking to 7.2.2 for a little while longer until hopefully the issue is resolved somehow.

**Comment by HashirLabs** [comment]
Hi [mateuszkwiecinski](https://github.com/mateuszkwiecinski)

any plan on fixing this issue??

**Comment by emartynov** [comment]
@HashirLabs @poorpeschino, please read the whole thread. There is a reason for the issue and a couple of proposed solutions. Particullary, https://github.com/usefulness/easylauncher-gradle-plugin/issues/349#issuecomment-1239054798, https://github.com/usefulness/easylauncher-gradle-plugin/issues/349#issuecomment-1274275555 and https://github.com/usefulness/easylauncher-gradle-plugin/issues/349#issuecomment-1277068235

**Comment by mateuszkwiecinski** [comment]
I'll share an update: I just released [v6.0.0](https://github.com/usefulness/easylauncher-gradle-plugin/releases/tag/v6.0.0) which helps to work around the issue (thanks to great contribution of [@kkris](https://github.com/usefulness/easylauncher-gradle-plugin/pull/369)).
It's possible `6.0.0` will work out of the box for you, but if you start observing a `"Error: Duplicate resources."` error, you can follow the workaround mentioned on [AGP issue page](https://issuetracker.google.com/issues/244063664#comment5) and add `android.disableResourceValidation=true` to `gradle.properties`. This should make the plugin work again 🎉 
If my observations are correct, AGP starting from 8.0 won't need the `android.disableResourceValidation=true` workaround.

**Comment by keith30xi** [comment]
We have the icons in a theme module.  That module also includes the easylauncher plugin. After upgrading, we no longer are seeing the generated icon.  We can see it in the build folder but the application module is not able to reference it. I looked at samples to see if we are missing something. 

Based on https://github.com/usefulness/easylauncher-gradle-plugin/blob/master/sample/gradle.properties, we added the following to our project.  Unfortunately it also made no difference.

```
android.disableResourceValidation=true
android.nonTransitiveRClass=true
```
